handler/src/main/java/io/netty/handler/pcap/PCapWriter.java

```diff
@@ -50,13 +48,10 @@
     void writePacket(ByteBuf packetHeaderBuf, ByteBuf packet) throws IOException {
         long currentTime = System.currentTimeMillis();
 
-        String microsecond = sdf.format(new Date(currentTime));
-        microsecond = microsecond.substring(microsecond.indexOf(".") + 1);
-
         PcapHeaders.writePacketHeader(
                 packetHeaderBuf,
-                currentTime,
-                Integer.parseInt(microsecond),
+                (int) TimeUnit.MILLISECONDS.toSeconds(currentTime),
+                (int) currentTime % 1000,
```

Comment:
This value is still in milliseconds, but looks like pcap wants microseconds. It could cause confusion to people using this, if we're off by a factor 1.000.

Reply:
Lol, need spoon feeding again. Haha. I tried but value went into negative. Help :p

Reply:
```java
import java.text.SimpleDateFormat;
import java.util.Date;

public class UIPF {
    private static final SimpleDateFormat sdf = new SimpleDateFormat("SSSSSSSSS");
    public static void main(String[] args) {
        long currentTime = System.currentTimeMillis();

        String microsecond = sdf.format(new Date(currentTime));
        microsecond = microsecond.substring(microsecond.indexOf(".") + 1);

        System.out.println(microsecond);
        System.out.println(currentTime % 1000);
    }
}
```

Output:
```
000000711
711
```

Considering zero in the prefix is not important, we're good. What do you say?

Reply:
Finally, after lots of tries. Figured it out.